File: CS 1450 Object Oriented Programming/Finals/Code/finals_4.py
class Hotel():

	def __init__(self, hotel_name):
		self.hotel_name = hotel_name
		self.hotel_guests_list = []

	def add_guest(self, guest_name):
		self.hotel_guests_list.append(guest_name)
		print(guest_name, "was checked in. Welcome!")

	def check_out_guest(self, guest_name):
		self.hotel_guests_list.remove(guest_name)
		print(guest_name, "was checked out. Good bye!")

	def get_list(self):
		return self.hotel_guests_list

hotel1 = Hotel("Hilton")
hotel1.add_guest("Patrik")
hotel1.add_guest("Phillip")
hotel1.add_guest("Erick")
print("Current guest list:", hotel1.get_list())
hotel1.check_out_guest("Erick")

File: CS 3140 The C++ Language/Finals/Code/finals_4.cpp
#include <iostream>
#include <string>
#include <list>
using namespace std;

class Hotel
{
private:
	string name;
	int num_guests = 0;

public:
	Hotel(string hotel_name)
	{
		name = hotel_name;
	}

	void add_guest(string guest_name)
	{
		num_guests += 1;
		cout << "The guest named " << guest_name << " has been succesfully checked in. Welcome!" << endl;		
	}

	void check_out_guest(string guest_name)
	{
		num_guests -= 1;
		cout << "The guest named " << guest_name << " has been succesfully checked out. Good bye!" << endl;	
	}

	int get_numberOfGuests()
	{
		
		return num_guests;
	}

};	

int main()
{
	Hotel hotel("Hilton");

	hotel.add_guest("Patrik");
	hotel.add_guest("Billy");
	hotel.add_guest("Jeff");
	hotel.add_guest("Jonah");

	cout << "The current number of guests is: " << hotel.get_numberOfGuests() << endl;

	hotel.check_out_guest("Jonah");

	cout << "The current number of guests is: " << hotel.get_numberOfGuests() << endl;

	return 0;
}
